etc/generate-envoy-config/envoy.jsonnet

```diff
@@ -0,0 +1,196 @@
+      route: {
+        cluster: 'pachd-s3',
+        idle_timeout: '600s',
+        timeout: '604800s',
+      },
+    }; [
+      base {  // S3v4
+        match: {
+          prefix: '/',
+          headers: [
+            {
+              name: 'authorization',
+              string_match: {
+                prefix: 'AWS4-HMAC-SHA256',
+              },
+            },
+          ],
+        },
+      },
+      base {  // S3v2
+        match: {
+          prefix: '/',
+          headers: [
+            {
+              name: 'authorization',
+              string_match: {
+                prefix: 'AWS ',
+              },
+            },
+          ],
+        },
+      },
+    ],
+  },
+  'pachd-identity': {
+    internal_port: 1658,
+    external_port: 30658,
+    service: 'pachd-proxy-backend',
+    routes: [
+      {
+        match: {
+          prefix: '/dex',
+        },
+        route: {
+          cluster: 'pachd-identity',
+          idle_timeout: '60s',
+          timeout: '60s',
+        },
+      },
+    ],
+    health_check: {
+      healthy_threshold: 1,
+      http_health_check: {
+        host: 'localhost',
+        path: '/dex/.well-known/openid-configuration',
+      },
+      interval: '30s',
+      timeout: '10s',
+      unhealthy_threshold: 2,
+      no_traffic_interval: '10s',
+      no_traffic_healthy_interval: '10s',
+    },
+  },
+  'pachd-oidc': {
+    internal_port: 1657,
+    external_port: 30657,
+    service: 'pachd-proxy-backend',
+    routes: [
+      {
+        match: {
+          prefix: '/authorization-code/callback',
+        },
+        route: {
+          cluster: 'pachd-oidc',
+          idle_timeout: '60s',
+          timeout: '60s',
+        },
+      },
+    ],
+  },
+  console: {
+    internal_port: 4000,
+    external_port: 4000,
+    service: 'console-proxy-backend',
+    routes: [
+      {
+        match: {
+          prefix: '/',
+        },
+        route: {
+          cluster: 'console',
+          idle_timeout: '600s',
+          timeout: '3600s',
+          upgrade_configs: [
+            {
+              enabled: true,
+              upgrade_type: 'websocket',
+            },
+          ],
+        },
+      },
+    ],
+    health_check: {
+      healthy_threshold: 1,
+      http_health_check: {
+        host: 'localhost',  // This is just the value of the Host: header, not something to connect to.
+        path: '/',
+      },
+      interval: '30s',
+      timeout: '10s',
+      unhealthy_threshold: 2,
+      no_traffic_interval: '10s',
+      no_traffic_healthy_interval: '10s',
+    },
+  },
+  'pachd-metrics': {
+    internal_port: 1656,
+    external_port: 30656,
+    service: 'pachd-proxy-backend',
+    routes: [
+      {
+        match: {
+          prefix: '/',
+        },
+        route: {
+          cluster: 'pachd-metrics',
+          idle_timeout: '60s',
+          timeout: '60s',
+        },
+      },
+    ],
+  },
+};
+
+// This is the generation code.
```

Comment:
how many threads is envoy running?

Reply:
I don't make any effort to set it.  Envoy is largely event-based: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/intro/threading_model